src/main/java/athleticli/commands/diet/DeleteDietCommand.java

```diff
@@ -0,0 +1,42 @@
+package athleticli.commands.diet;
+
+import athleticli.commands.Command;
+import athleticli.data.Data;
+import athleticli.data.diet.Diet;
+import athleticli.data.diet.DietList;
+import athleticli.exceptions.AthletiException;
+import athleticli.ui.Message;
+
+/**
+ * Executes the add diet commands provided by the user.
+ */
+public class DeleteDietCommand extends Command {
+    private final int index;
+
+    /**
+     * Constructor for AddDietCommand.
+     *
+     * @param diet Diet to be added.
+     */
+    public DeleteDietCommand(int index) {
+        this.index = index;
+    }
+
+    /**
+     * Updates the diet list.
+     *
+     * @param data The current data containing the diet list.
+     * @return The message which will be shown to the user.
+     */
+    public String[] execute(Data data) throws AthletiException {
+        if (index > data.getDiets().size() || index < 1) {
+            throw new AthletiException(Message.MESSAGE_INVALID_DIET_INDEX);
+        }
+        DietList dietList = data.getDiets();
+        int size = dietList.size();
+        Diet oldDiet = dietList.get(index - 1);
```

Comment:
Would be preferable to rearrange this to the top of this method signature such that the exception checker can use index > size for better readability

Reply:
Good idea. Resolved.